packages/migration/src/migrations/user-mgnt/20240628124653-remove-userroles.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+
+import { Db, MongoClient } from 'mongodb'
+
+export const up = async (db: Db, client: MongoClient) => {
+  const session = client.startSession()
+  try {
+    await session.withTransaction(async () => {
+      await db
+        .collection('users')
+        .updateMany(
+          { username: 'o.admin' },
+          { $set: { role: 'SUPER_ADMIN' }, $unset: { systemRole: '' } }
+        )
+      await db
+        .collection('users')
+        .aggregate([
+          {
+            $lookup: {
+              from: 'userroles',
+              localField: 'role',
+              foreignField: '_id',
+              as: 'roleDetails'
+            }
+          },
+          {
+            $unwind: '$roleDetails'
+          },
+          {
+            $addFields: {
+              role: {
+                $toUpper: {
+                  $replaceAll: {
+                    input: { $arrayElemAt: ['$roleDetails.labels.label', 0] },
+                    find: ' ',
+                    replacement: '_'
+                  }
+                }
+              }
+            }
+          },
+          {
+            $project: {
+              roleDetails: 0
+            }
+          },
+          {
+            $merge: {
+              into: 'users',
+              whenMatched: 'merge',
+              whenNotMatched: 'fail'
+            }
+          }
+        ])
+        .toArray()
```

Comment:
@rikukissa @Nil20 This doesn't look right. If I am not wrong, we just need to copy `user.systemRole` to `user.role` and then delete `user.systemRole`. The `userroles` collection has the labels that were created through the UI/seeding, which could be anything

Reply:
I get it now why you did it like that. This way we are not losing the newly created roles. In that case we just need to make sure `roles.csv` is properly transformed into `roles.json`. maybe even create a script for it if possible